openfasoc/generators/gdsfactory-gen/glayout/primitives/fet.py

```diff
@@ -422,7 +423,41 @@ def nmos(
         )
         tapring_ref = nfet << ringtoadd
         nfet.add_ports(tapring_ref.get_ports_list(),prefix="guardring_")
-    return rename_ports_by_orientation(nfet).flatten()
+
+    component = rename_ports_by_orientation(nfet).flatten()
+
+    # add spice netlist
+    num_dummies = 0
```

Comment:
wrap all the netlist logic into another function and do a single add at the very end. 

`nmos.info["netlist"]=nmos_netlist(params....)`